### PR TITLE
Consensus fixes

### PR DESCRIPTION
**Pull request by roman-khimov**
### Problem

Testnet CN node.

### Solution

Fix multiple problems and make it propose something that other nodes can accept.

### CODE REVIEW

File: pkg/consensus/consensus.go [hunk 1]
```diff
@@ -204,6 +204,21 @@ func (s *service) eventLoop() {
 			s.dbft.OnReceive(&msg)
 		case tx := <-s.transactions:
 			s.dbft.OnTransaction(tx)
+		case <-s.blockEvents:
+			// Drain events if there is more than one event queued.
+		drain:
+			for {
+				select {
+				case <-s.blockEvents:
```
Comment (fyrchik) [inline pkg/consensus/consensus.go]: What about removing this loop and using `chan struct{}` of capacity 1 with non-blocking select on the send side?
It seems to have less overhead.

File: config/config.go [hunk 2]
```diff
@@ -59,6 +59,13 @@ type (
 		FreeGasLimit util.Fixed8 `yaml:"FreeGasLimit"`
 		// SaveStorageBatch enables storage batch saving before every persist.
 		SaveStorageBatch bool `yaml:"SaveStorageBatch"`
+		// Maximum number of low priority transactions accepted into block.
+		MaxFreeTransactionsPerBlock int `yaml:"MaxFreeTransactionsPerBlock"`
+		// Maximum size of low priority transaction.
```
Comment (fyrchik) [inline config/config.go]: `... in bytes.` ?

File: pkg/core/util.go [hunk 3]
```diff
@@ -13,6 +13,17 @@ import (
 	"github.com/CityOfZion/neo-go/pkg/vm/opcode"
 )
 
+var (
+	// GoverningTokenTX represents transaction that is used to create
+	// governing (NEO) token. It's a part of the genesis block.
+	GoverningTokenTX *transaction.Transaction
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/core/util.go]: One thing I'm really missing in Go is proper constant types. It wouldn't be a problem if I could export some `const *transaction.Transaction`. At the same time I think that whole transaction can be more useful than its hash. For example, there is Amount and Precision encoded there, although one might argue that we can get it from `AssetState`. So maybe it's still worth hiding the transaction itself.

File: pkg/consensus/consensus.go [hunk 4]
```diff
@@ -411,10 +411,25 @@ func (s *service) getVerifiedTx(count int) []block.Transaction {
...
 		res[i+1] = txx[i]
+		netFee += s.Config.Chain.NetworkFee(txx[i])
 	}
 
+	var txOuts []transaction.Output
+	if netFee != 0 {
+		sh := s.wallet.GetChangeAddress()
+		if sh.Equals(util.Uint160{}) {
+			pk := s.dbft.Pub.(*publicKey)
+			sh = hash.Hash160(pk.GetVerificationScript())
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/consensus/consensus.go]: Yeah, thought about that too, but decided not to touch it at the moment.

**Pull request merged**